cEP-0017.md

```diff
@@ -0,0 +1,366 @@
+example is shown below:
+
+```
+tasks:
+  - name: ensure apache is at the latest version
+    yum:
+      name: httpd
+      state: latest
+  - name: write the apache config file
+    template:
+      src: /srv/httpd.j2
+      dest: /etc/httpd.conf
+    notify:
+    - restart apache
+  - name: ensure apache is running (and enable it at boot)
+    service:
+      name: httpd
+      state: started
+      enabled: yes
+```
+
+Ansible provides numerous
+[packaging modules](http://docs.Ansible.com/Ansible/latest/modules/list_of_packaging_modules.html)
+that can be used to install dependencies of `Bears`.
+Another advantage of Ansible is its ability to do a large amount of fact gathering
+from the user's machine and provide relevant information such as `Ansible_distribution` and
+`Ansible_pkg_mgr`. We can use these gathered facts to automatically decide in certain cases
+ such as shown below.
+
+```
+- name: install basic package
+  action: >
+    {{ ansible_pkg_mgr }} name=package_name
+```
+
+Other than this, Ansible also provides options to gain root access for installing packages
+and printing verbose output. Ansible also allows filtering tasks based on tags.
+This can be used to install bears for a specific language, certain capability etc
+if the appropriate tags are added.
+
+## Proposed Changes
+
+### Procedure for the installation of the bears using cib
+
+1. Identify the valid bear names that need to be installed as specified by the
+   user on the command line.
+2. Install bear on the valid bear names. We will run commands from `CondaRequirement`
+   to check whether the package is uploaded to `Anaconda.org` and then check whether
+   it can be installed from `PyPi` using `PipRequirement`.
+   - This will require to create conda packages for bears that have different
+     installation instructions for each distribution. This is done by making `bld.bat`
+     file that has installation instruction for Windows and similarly a `bld.sh` file
+     that has installation instruction for Linux.
+3. After installing the bear we proceed to installing its requirements. We call
+   methods from the requirement classes to do this.
+   - Add new Requirement classes to support different dependencies from various linters.
+   - The DistributionRequirement class needs to be split into different classes for each
+     package manager and call its method to introduce more consistency.
+4. We then proceed to install bear dependencies such as for `ClangCloneDetectionBear` by
+   running `install_bears` function for each dependency bear.
+
+### Conda Packaging
+
+1. Create a new repository called `coala-bear packages`.
+2. This repository will have a folder called `conda packages` that contains conda packages for
+   all the bears that require different installation instructions on each distribution. Each
+   conda package requires 3 files i.e. `meta.yaml`, `build.sh`, `bld.bat`.
```

Comment:
Above (line 90) you have mentioned `bld.sh` and here it is `build.sh`. You should stick to one name.

Reply:
(fixed)